dali/kernels/imgproc/color_manipulation/equalize/equalize.h

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_EQUALIZE_EQUALIZE_H_
+#define DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_EQUALIZE_EQUALIZE_H_
+
+#include <vector>
+
+#include "dali/core/common.h"
+#include "dali/kernels/imgproc/color_manipulation/equalize/hist.h"
+#include "dali/kernels/imgproc/color_manipulation/equalize/lookup.h"
+#include "dali/kernels/imgproc/color_manipulation/equalize/lut.h"
+#include "dali/kernels/kernel.h"
+#include "include/dali/core/backend_tags.h"
+#include "include/dali/core/tensor_view.h"
+
+namespace dali {
+namespace kernels {
+namespace equalize {
+
+struct EqualizeKernelGpu {
+  static constexpr int hist_range = 256;
+
+  /**
+   * @brief Performs per-channel equalization.
```

Comment:
This doesn't seem to do per-channel equalization, but equalization on a single-channel input.

Reply:
The input can have multiple channels (the second extent), each of them will get different histogram and lookup tables.